src/inatinqperf/benchmark/configuration.py

```diff
@@ -0,0 +1,62 @@
+"""Modules with classes for loading configurations with Pydantic validation."""
+
+from pathlib import Path
+from typing import Any
+
+from pydantic import BaseModel, PositiveInt
+
+
+class DatasetConfig(BaseModel):
+    """Dataset definition within the main configuration."""
+
+    dataset_id: str
```

Comment:
Use `pydantic`'s `constr` to refine what is valid for an id (at the very least, it should be a non-empty string). Apply the same comment to add ids in this file.